docs/src/main/tut/docs/stack/README.md

```diff
@@ -0,0 +1,260 @@
+  new StockPersistence.Handler[Stack] {
+    def checkQuantityAvailable(cultivar: String): Stack[Int] =
+      (cultivar.toLowerCase match {
+        case "granny smith" => 150
+        case "jonagold"     => 200
+        case _              => 25
+      }).pure[Stack]
+
+
+    def registerOrder(order: Order): Stack[Unit] =
+      Kleisli.lift(WriterT.lift(Task.delay(println(s"Register $order"))))
+  }
+```
+
+Below are some type class instances for `Kleisli` which are currently not yet provided by Cats.
+
+```tut:book
+import cats.data.Kleisli
+
+// MonadError for Kleisli is on its way already
+// https://github.com/typelevel/cats/pull/1543
+
+import cats.{Monad, MonadError, MonadWriter}
+
+trait KleisliMonad[F[_], A] {
+  implicit val F: Monad[F]
+
+  def pure[B](x: B): Kleisli[F, A, B] =
+    Kleisli.pure[F, A, B](x)
+
+  def flatMap[B, C](fa: Kleisli[F, A, B])(f: B => Kleisli[F, A, C]): Kleisli[F, A, C] =
+    fa.flatMap(f)
+
+  def tailRecM[B, C](b: B)(f: B => Kleisli[F, A, Either[B, C]]): Kleisli[F, A, C] =
+    Kleisli[F, A, C]({ a => F.tailRecM(b) { f(_).run(a) } })
+}
+
+trait KleisliMTL1 {
+  implicit def kleisliMonadError[F[_], A, E](implicit ME: MonadError[F, E]): MonadError[Kleisli[F, A, ?], E] = 
+    new MonadError[Kleisli[F, A, ?], E] with KleisliMonad[F, A] {
+      implicit val F = ME
+
+      def raiseError[B](e: E): Kleisli[F, A, B] = Kleisli(_ => ME.raiseError(e))
+
+      def handleErrorWith[B](kb: Kleisli[F, A, B])(f: E => Kleisli[F, A, B]): Kleisli[F, A, B] = Kleisli { a: A =>
+        ME.handleErrorWith(kb.run(a))((e: E) => f(e).run(a))
+      }
+    }
+}
+
+object KleisliMTL extends KleisliMTL1 {
+  implicit def kleisliMonadWriter[F[_], A, L](implicit MW: MonadWriter[F, L]): MonadWriter[Kleisli[F, A, ?], L] = 
+    new MonadWriter[Kleisli[F, A, ?], L] with KleisliMonad[F, A] {
+      implicit val F = MW
+
+      def writer[B](bw: (L, B)): Kleisli[F, A, B] =
+        Kleisli.lift(MW.writer(bw))
+
+      def listen[B](fa: Kleisli[F, A, B]): Kleisli[F, A, (L, B)] =
+        Kleisli(a => MW.listen(fa.run(a)))
+
+      def pass[B](fb: Kleisli[F, A, (L => L, B)]): Kleisli[F, A, B] =
+        Kleisli(a => MW.pass(fb.run(a)))
+
+      override def tell(l: L): Kleisli[F, A, Unit] = 
+        Kleisli.lift(MW.tell(l))
+    }
+}
+```
+
+
+We can now create a Freestyle program by specifying the type in `processOrder` as `App.Op`.
+
+```tut:book
+val program: FreeS[App.Op, String] =
+  processOrder[App.Op](Order(50, "granny smith", customerId1))
+```
+
+With the persistence algebras handlers in place and with the right imports for the Freestyle effect algebras, we can execute to program resulting in `Stack[String]`.
+
+```tut:book
+import freestyle.effects.error.implicits._
+import wr.implicits._
+import rd.implicits._
+import KleisliMTL._
+
+val result: Stack[String] = program.exec[Stack]
+```
+
+We can run our `Stack` by supplying a `Config` value, running the `WriterT` and running the `Task`.
```

Comment:
We should end with a conclusion of the advantages of using Freestyle here to create programs descriptions that are interpretable and how evaluation and effects are pushed toward the edge etc. It may not be immediately obvious to someone why this may be a better approach over imperatively writing the same logic or even using vanilla scala.